# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/BrowsingContext.h
# Commit: 531e69180f15
# Full Hash: 531e69180f15e79c2449e747c0204b83e3a1aee7
# Author: Andreas Farre <farre@mozilla.com>
# Date: 2019-07-03 04:09:04
# Regressor Bug: 1561899
# File Overlap Count: 2
# Description:
#   Bug 1561899 - Add mIsDiscarded and use that when detaching. r=nika
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D36195
# ==============================================================================

diff -r 9c6775d713e8 -r 531e69180f15 docshell/base/BrowsingContext.h
--- a/docshell/base/BrowsingContext.h	Tue Jul 02 17:15:18 2019 +0000
+++ b/docshell/base/BrowsingContext.h	Tue Jul 02 20:48:13 2019 +0000
@@ -380,6 +380,9 @@
 
   bool IsActive() const;
 
+  // Removes the context from its group and sets mIsDetached to true.
+  void Unregister();
+
   friend class ::nsOuterWindowProxy;
   friend class ::nsGlobalWindowOuter;
   // Update the window proxy object that corresponds to this browsing context.
@@ -461,6 +464,10 @@
   // process? This may be true with a null mDocShell after the Window has been
   // closed.
   bool mIsInProcess : 1;
+
+  // Has this browsing context been discarded? BrowsingContexts should
+  // only be discarded once.
+  bool mIsDiscarded : 1;
 };
 
 /**
